docs/sop/PRODUCTION_DEPLOY.md

```diff
@@ -0,0 +1,344 @@
+$ mkdir gigascience
+```
+
+Change directory to the `gigascience` folder:
+```
+$ cd gigascience
+```
+
+Download the gigascience/gigadb-website repo from Github:
+```
+$ git clone https://github.com/gigascience/gigadb-website.git
+```
+
+Change directory into the repo:
+```
+$ cd gigadb-website
+```
+
+Check you are tracking the `origin` repository:
+```
+$ git remote -v
+origin  https://github.com/gigascience/gigadb-website.git (fetch)
+origin  https://github.com/gigascience/gigadb-website.git (push)
+```
+
+Switch to the `develop` branch:
+```
+$ git checkout develop
+```
+
+Check you have the latest code for the `develop` branch:
+```
+$ git log
+```
+
+If not, then get the latest code:
+```
+$ git fetch origin
+$ git rebase origin/develop
+```
+
+Check you can deploy GigaDB in your `dev` environment:
+```
+# Create .env file
+$ cp ops/configuration/variables/env-sample .env
+
+# Edit .env file with particular focus on the variables below:
+GITLAB_PRIVATE_TOKEN=<your_gitlab_private_token>
+
+REPO_NAME="gigadb-website"
+GROUP_VARIABLES_URL="https://gitlab.com/api/v4/groups/gigascience/variables?per_page=100"
+FORK_VARIABLES_URL="https://gitlab.com/api/v4/groups/3506500/variables"
+PROJECT_VARIABLES_URL="https://gitlab.com/api/v4/projects/gigascience%2Fupstream%2F$REPO_NAME/variables"
+
+# Execute up.sh
+$ ./up.sh
+```
+
+Check your `dev` GigaDB website is available at http://gigadb.gigasciencejournal.com:9170 
+in your browser.
+
+### Provision AWS infrastructure for `staging.gigadb.org` using Terraform and Ansible
+
+Ensure you have a database back up file that Ansible can use to restore a
+PostgreSQL database on the AWS Relational Database Service. For example, copy
+a file from your `dev` gigadb-website repo:
+```
+$ cp <path_to>/pli888/gigadb-website/gigadb/app/tools/files-url-updater/sql/gigadbv3_20210929_v9.3.25.backup <path_to>/gigascience/gigadb-website/gigadb/app/tools/files-url-updater/sql
+```
+
+Before you are able to create a `live` deployment, you must first deploy a 
+`staging` environment.
+
+And you have to make sure that terraform state files exist in [gitlab terraform page](https://gitlab.com/gigascience/upstream/gigadb-website/-/terraform).
+You can click to download the [shared Terraform state for Upstream's staging environment](https://gitlab.com/api/v4/projects/11385199/terraform/state/staging_infra/versions/144)
+for checking in the latter staging deployment steps
+
+Change directory to the `envs` folder:
+```
+$ cd ops/infrastructure/envs
+```
+
+Change directory to the `staging` folder:
+```
+$ cd staging
+```
+
+Copy terraform files to `staging` environment:
+```
+# this step will retrieve terraform state file from the gitlab
+$ ../../../scripts/tf_init.sh --project gigascience/upstream/gigadb-website --env staging
+
+You need to specify the path to the ssh private key to use to connect to the EC2 instance: 
+~/.ssh/id-rsa-aws-hk-gigadb.pem
+
+You need to specify your GitLab username:
+pli888 | rija | kencho51
```

Comment:
**issue**: no need to put this, his is security issue
**suggestion**: remove usernames